kvbc/src/categorization/block_merkle_category.cpp

```diff
@@ -541,6 +541,35 @@ std::pair<SetOfKeyValuePairs, KeysVector> BlockMerkleCategory::rewriteAlreadyPru
   return std::make_pair(merkle_blocks_to_rewrite, merkle_blocks_to_delete);
 }
 
+std::vector<std::string> BlockMerkleCategory::getBlockStaleKeys(BlockId block_id, const BlockMerkleOutput& out) {
+  std::vector<Hash> hash_stale_keys;
+  auto [hashed_keys, latest_versions] = getLatestVersions(out);
+  for (auto i = 0u; i < hashed_keys.size(); i++) {
+    auto& tagged_version = latest_versions[i];
+    auto& hashed_key = hashed_keys[i];
+    ConcordAssert(tagged_version.has_value());
+    ConcordAssertLE(block_id, tagged_version->version);
+
+    if (block_id == tagged_version->version) {
+      if (tagged_version->deleted) {
+        // The latest version is a tombstone. We can delete the key and version.
```

Comment:
I think the `we can delete the key and version` part of the comment is not relevant here.

Reply:
fixed